code/lib/instrumenter/src/instrumenter.ts

```diff
@@ -112,6 +114,10 @@ export class Instrumenter {
       isPlaying?: boolean;
       isDebugging?: boolean;
     }) => {
+      if (!storyId) {
+        return;
+      }
+
       const state = this.getState(storyId);
```

Comment:
I think this might break the code. It seems like getState, setState and sync are able to handle a null value, so it might be better to just loose the type definition of those methods, so that the initialState stays being used whenever `storyId` is null.

Reply:
It seems like setting an `undefined` or `null` as a `storyId` could cause unintended consequences?

https://github.com/storybookjs/storybook/blob/7e573248c43ade20431b79e2da59c93f1cd8ca82/code/lib/instrumenter/src/instrumenter.ts#L241

This line would set "null" or "undefined" as the property name.

Reply:
You are right. After the reviewing the code a bit more, I think we should make the `storyId` parameter non-nullable. We really expect this function to be called with a storyId. Could you try that?

Reply:
This worked.  Updated PR.